### DISCUSSION TITLE
Send Frames - Custom - Use dbc

### DISCUSSION THREAD

**Discussion by simoneruffini** [collin80/SavvyCAN#664] [Q&A] [answered]
In the Send Frames - Custom window, is there a way to send messages and signals using the loaded dbc instead of typing the raw data by hand?

**Comment by collin80** [comment, marked as answer]
No, but that's an excellent idea. I want to incorporate DBC signals into more places in the program. It should be possible to use them nearly anywhere frames can be sent from.

**Reply by simoneruffini** [reply]
Is it hard to implement? This feature could be really something because right now there are no GUI programs that let you do that in the linux world (from what i know) 
BusMaster is the only FOSS that does what savvycan do, but is windows only and no more maintained.

**Reply by collin80** [reply]
I'm implementing it currently. It's a bit of a change to the custom frame sender and I'm making various parts easier and/or better so the work is slow but it is progressing. I expect I should have it working by next week as I'd like to use it then.

**Reply by simoneruffini** [reply]
I was thinking something like busmaster does:
![image](https://github.com/collin80/SavvyCAN/assets/50401154/fb75fc62-783f-4e87-9aa9-71e23fbec349)
Essentially you have two sections, the main one for sending the message (top section of the image) that does the same as SavvyCAN.
But you have the message column where you can select messages from a dbc file.
The main addition instead is the bottom section visible in the image, where: when selecting a message (a row in the top section) you are presented with the list of signals in the message that you can toggle or change correctly by dbc specification.

Moreover if you are working on a branch can you share which one? Maybe I can be of some help.

**Reply by MatheusCaltran** [reply]
Has this implementation been added yet?
